### ISSUE TITLE
Hosting fanmade/alternate/community/[put sth here] rulesets and datafiles

### ISSUE THREAD

**Issue by amis92** [BSData/catalogue-development#86]
## TLDR

We're going to focus on enabling every single person and organization to be able to host their files and have them easily accessible through app's Data Management interface. This may take time; if you're a software developer - read on to find out how you can contribute.

## The problem

There has been an increasing amount of requests for hosting new game systems on our website. We've had a policy to host only official game system repositories, but as with everything, the definition is not sufficient. Exceptions have been made due to various reasons. When exceptions are made, it's hard to force the rules going forward.

Some rulesets not published by miniature designers but by other entities are even printed and published in professional manner, to the "official" or "commercial" requirements are not working anway.

There's no actual reason for us to stop people from discovering and using these rulesets. The only reason we've filtered out some of them is that we tried to keep things clean. Fanmade stuff tends to quickly die (in comparison to game/miniature companies). As I was the only person in charge of that, I did not want to have the burden of keeping this in check on me as well.

Life is life, and in the last couple of years even official systems got replaced with other stuff (AoS). Ironic.

## The solution

To allow any interested party in publishing their datafiles simply and mostly on their own, and take off the liability of maintaining the repository access by BSData org (us), we need to **redesign** our indexing and distribution system.

This is no small task! But it's necessary to keep going forward. The system would involve a more decentralized serving of the datafile assets, with our web server being minimized to an indexer that gathers metadata and points in the right direction. The specifics and technicalities I'll leave out of this and instead reference BSData/BSData#243

**Comment by amis92** [comment]
We still refrain from hosting fanmade datafiles (or fanmade extensions).

However, we now have a nice guide on how to host your repositories more conveniently so that they're easily accessible to your audience via GitHub. Please give the following page from our wiki a read: https://github.com/BSData/catalogue-development/wiki/Help:-Hosting-repositories
